api/deploy.go

```diff
@@ -58,20 +58,16 @@ func (d *Deploy) SendFile() (err error) {
 	d.SetRawBody(body)
 	d.SetResponseReceiver(resp)
 	if err = d.DoCall(); err != nil {
-		return
-	}
-
-	code := d.StatusCode()
-
-	if code == http.StatusUnauthorized {
-		err = ErrUnauthorized
-	} else if code == http.StatusUnprocessableEntity {
-		err = ErrPayloadValidation
-	} else if code != http.StatusOK && code != http.StatusCreated {
-		err = ErrBadResponseStatus
-	}
-
-	if err != nil {
+		if errAPI, is := err.(*ErrAPI); is {
```

Comment:
As I understand, `errors.Is()` checks for value, and not type... since each new pointer of `*ErrAPI` would have its own possible value, asserting the underlying type behind the interface seems like the way to go.

https://golang.org/pkg/errors/#Is